### DISCUSSION TITLE
Make it easier to create a theme with default StyleBox and customize from here

### DISCUSSION THREAD

**Discussion by hsandt** [godotengine/godot-proposals#6861] [GUI]
I created a theme with a custom StyleBox, but noted that I lost the nice selection edge on focus around my buttons, even though I made my StyleBoxes based on the default values of StyleBoxFlat.

Default theme and styleboxes

![image](https://github.com/godotengine/godot-proposals/assets/1067772/9bee76bd-a7f4-450f-b685-061c8487232d)

Custom theme and styleboxes (just changed the font and minor params)

![image](https://github.com/godotengine/godot-proposals/assets/1067772/6537327f-6455-4d12-9f92-9858c0ee24a3)

I understand that default values are set to make sense codewise, and may differ from the default values used for Godot's default theme.

Is there a way to reset a StyleBox to these defaults then? Is the default even a StyleBoxFlat, or another type or StyleBox?

For now, I'll manualy re-add borders...

**Comment by Calinou** [comment]
> Is there a way to reset a StyleBox to these defaults then? Is the default even a StyleBoxFlat, or another type or StyleBox?

The default is a StyleBoxFlat, but with non-default parameters. The default StyleBoxFlat is an opaque gray box with no borders or rounded corners.

**Reply by hsandt** [reply]
OK, so it's really a problem of default preset vs raw default values as defined in class (which is generally filled with 0s, but apparently not everywhere, e.g. I can still see rounded bordered).

So it would be nice to be able to create a custom StyleBoxFlat using the default preset StyleBoxFlat as template. So people who only need to tweak one parameter don't have to redefine their new StyleBoxFlat resource to look like the default preset, then tweak their parameter.

**Comment by pineapplemachine** [comment]
Heads up to anyone who ends up here like me, trying to figure this out: This is possible now!

Steps:

If you don't have one already, create a new Theme resource somewhere in your project and double-click on it, to bring up the theme panel in the bottom of the editor:
![image](https://github.com/user-attachments/assets/766143eb-8e72-4a52-ac15-2effd498bd55)

In the theme panel, first click the eyedropper and then on a button:
![image](https://github.com/user-attachments/assets/e7c7dfce-8469-4121-a871-779c21253a68)
![image](https://github.com/user-attachments/assets/1c92b4c2-e882-40c0-ad5c-e809c7a86436)

On the right side of the theme panel, go to the styleboxes tab:
![image](https://github.com/user-attachments/assets/b5606d93-937d-4b5e-b509-2b2409ab3cc5)

Right click the style you want to use as a basis and click "Copy" in the context menu:
![image](https://github.com/user-attachments/assets/ccf6d833-81ba-418b-8a6b-7d9e46f1b7c4)

In your button's theme overrides settings, right click the style you want to edit and click "Paste":
![image](https://github.com/user-attachments/assets/23fffdfe-e2a2-45e1-9c30-29622a9c858e)

Now you can click on the StyleBox resource here to begin to edit it, using the default settings as a baseline:
![image](https://github.com/user-attachments/assets/6772ad43-66e9-4e7f-bc33-a97031aa21d3)

**Reply by BaconEggsRL** [reply]
Thank youuuu
(As an aside, if you are building a theme, just right click the existing default styleboxflat in the theme -> copy -> hit + sign to create new -> right click again -> paste)
